### ISSUE TITLE
will you add proxy-protocol support?

### ISSUE THREAD

**Issue by lootoos** [editso/fuso#35]
in order to be able to get normal IP addresses to the minecraft server.
in order for the server to be able to receive external IP addresses of players.
by the way, nginx supports proxy_protocol (https://imgbb.com/FnpFhPk), will you introduce such functionality in fuso?

**Comment by editso** [comment]
> in order to be able to get normal IP addresses to the minecraft server. in order for the server to be able to receive external IP addresses of players. by the way, nginx supports proxy_protocol (https://imgbb.com/FnpFhPk), will you introduce such functionality in fuso?

do you mean, forward external real ip to fuso client?

**Comment by editso** [comment]
> in order to be able to get normal IP addresses to the minecraft server. in order for the server to be able to receive external IP addresses of players. by the way, nginx supports proxy_protocol (https://imgbb.com/FnpFhPk), will you introduce such functionality in fuso?

Is this question the same as #33?

**Comment by lootoos** [comment]
> > in order to be able to get normal IP addresses to the minecraft server. in order for the server to be able to receive external IP addresses of players. by the way, nginx supports proxy_protocol (https://imgbb.com/FnpFhPk), will you introduce such functionality in fuso?
> 
> do you mean, forward external real ip to fuso client?

yes

**Comment by lootoos** [comment]
> > in order to be able to get normal IP addresses to the minecraft server. in order for the server to be able to receive external IP addresses of players. by the way, nginx supports proxy_protocol (https://imgbb.com/FnpFhPk), will you introduce such functionality in fuso?
> 
> Is this question the same as #33?

no, but the question is similar, ip forward is different from proxy-protocol

**Comment by editso** [comment]
> > > in order to be able to get normal IP addresses to the minecraft server. in order for the server to be able to receive external IP addresses of players. by the way, nginx supports proxy_protocol (https://imgbb.com/FnpFhPk), will you introduce such functionality in fuso?
> > 
> > 
> > Is this question the same as #33?
> 
> no, but the question is similar, ip forward is different from proxy-protocol

You can try to modify [handshake#L138](https://github.com/editso/fuso/blob/refactor/dev/src/net/penetrate/handshake/mod.rs#L138) to achieve

**Comment by lootoos** [comment]
> > > > чтобы иметь возможность получить нормальные IP-адреса для сервера minecraft. для того, чтобы сервер мог получать внешние IP-адреса игроков. кстати, nginx поддерживает proxy_protocol ( https://imgbb.com/FnpFhPk ), в fuso будете внедрять такой функционал?
> > > 
> > > 
> > > Этот вопрос такой же, как [№ 33](https://github.com/editso/fuso/issues/33) ?
> > 
> > 
> > нет, но вопрос аналогичный, ip forward отличается от proxy-протокола
> 
> Вы можете попробовать изменить [рукопожатие # L138](https://github.com/editso/fuso/blob/refactor/dev/src/net/penetrate/handshake/mod.rs#L138) для достижения

no entiendo el lenguaje "rust"

**Comment by lootoos** [comment]
> > > > in order to be able to get normal IP addresses to the minecraft server. in order for the server to be able to receive external IP addresses of players. by the way, nginx supports proxy_protocol (https://imgbb.com/FnpFhPk), will you introduce such functionality in fuso?
> > > 
> > > 
> > > Is this question the same as #33?
> > 
> > 
> > no, but the question is similar, ip forward is different from proxy-protocol
> 
> You can try to modify [handshake#L138](https://github.com/editso/fuso/blob/refactor/dev/src/net/penetrate/handshake/mod.rs#L138) to achieve

I don't understand the language "rust"

**Comment by editso** [comment]
> > > > > in order to be able to get normal IP addresses to the minecraft server. in order for the server to be able to receive external IP addresses of players. by the way, nginx supports proxy_protocol (https://imgbb.com/FnpFhPk), will you introduce such functionality in fuso?
> > > > 
> > > > 
> > > > Is this question the same as #33?
> > > 
> > > 
> > > no, but the question is similar, ip forward is different from proxy-protocol
> > 
> > 
> > You can try to modify [handshake#L138](https://github.com/editso/fuso/blob/refactor/dev/src/net/penetrate/handshake/mod.rs#L138) to achieve
> 
> no entiendo el lenguaje "rust"

I can implement it, but it will take a little time

**Comment by lootoos** [comment]
> > > > > > in order to be able to get normal IP addresses to the minecraft server. in order for the server to be able to receive external IP addresses of players. by the way, nginx supports proxy_protocol (https://imgbb.com/FnpFhPk), will you introduce such functionality in fuso?
> > > > > 
> > > > > 
> > > > > Is this question the same as #33?
> > > > 
> > > > 
> > > > no, but the question is similar, ip forward is different from proxy-protocol
> > > 
> > > 
> > > You can try to modify [handshake#L138](https://github.com/editso/fuso/blob/refactor/dev/src/net/penetrate/handshake/mod.rs#L138) to achieve
> > 
> > 
> > no entiendo el lenguaje "rust"
> 
> I can implement it, but it will take a little time

is it also possible to implement this for the udp protocol?

**Comment by editso** [comment]
> > > > > > > in order to be able to get normal IP addresses to the minecraft server. in order for the server to be able to receive external IP addresses of players. by the way, nginx supports proxy_protocol (https://imgbb.com/FnpFhPk), will you introduce such functionality in fuso?
> > > > > > 
> > > > > > 
> > > > > > Is this question the same as #33?
> > > > > 
> > > > > 
> > > > > no, but the question is similar, ip forward is different from proxy-protocol
> > > > 
> > > > 
> > > > You can try to modify [handshake#L138](https://github.com/editso/fuso/blob/refactor/dev/src/net/penetrate/handshake/mod.rs#L138) to achieve
> > > 
> > > 
> > > no entiendo el lenguaje "rust"
> > 
> > 
> > I can implement it, but it will take a little time
> 
> is it also possible to implement this for the udp protocol?

I think it's possible

**Comment by lootoos** [comment]
> > > > > > in order to be able to get normal IP addresses to the minecraft server. in order for the server to be able to receive external IP addresses of players. by the way, nginx supports proxy_protocol (https://imgbb.com/FnpFhPk), will you introduce such functionality in fuso?
> > > > > 
> > > > > 
> > > > > Is this question the same as #33?
> > > > 
> > > > 
> > > > no, but the question is similar, ip forward is different from proxy-protocol
> > > 
> > > 
> > > You can try to modify [handshake#L138](https://github.com/editso/fuso/blob/refactor/dev/src/net/penetrate/handshake/mod.rs#L138) to achieve
> > 
> > 
> > no entiendo el lenguaje "rust"
> 
> I can implement it, but it will take a little time

ok

**Comment by lootoos** [comment]
Proxy Protocol to send user’s real IP to local services.
